src/views/catalog/templatescatalog/components/TemplatesCatalogDrawer/TemplatesCatalogDrawerFooter.tsx

```diff
@@ -32,16 +37,31 @@ type TemplateCatalogDrawerFooterProps = {
   template: undefined | V1Template;
 };
 
-export const TemplatesCatalogDrawerFooter: React.FC<TemplateCatalogDrawerFooterProps> = ({
+export const TemplatesCatalogDrawerFooter: FC<TemplateCatalogDrawerFooterProps> = ({
   namespace,
   onCancel,
   template,
 }) => {
   const { t } = useKubevirtTranslation();
   const { isBootSourceAvailable, loaded: bootSourceLoaded } = useVmTemplateSource(template);
-  const [authorizedSSHKeys, , userSettingsLoaded] = useKubevirtUserSettings('ssh');
+  const [authorizedSSHKeys, updateAuthorizedSSHKeys, userSettingsLoaded] =
+    useKubevirtUserSettings('ssh');
   const { loaded: loadedRHELSubscription, subscriptionData } = useRHELAutomaticSubscription();
   const [processedTemplate, processedTemplateLoaded] = useProcessedTemplate(template, namespace);
+  const [, , loadError] = useK8sWatchResource<IoK8sApiCoreV1Secret>(
+    authorizedSSHKeys?.[namespace] && {
+      groupVersionKind: getGroupVersionKindForModel(SecretModel),
+      isList: false,
+      name: authorizedSSHKeys?.[namespace],
+      namespace,
+    },
+  );
+
+  useEffect(() => {
+    if (loadError && authorizedSSHKeys?.[namespace]) {
```

Comment:
can u please add a comment if the secret isn't exisit removing the secret from namespace ?